src/api/app/controllers/webui/watched_items_controller.rb

```diff
@@ -28,10 +28,9 @@ def toggle_watched_item
   private
 
   def set_watchable
-    @watchable = if params[:project_name] && params[:package_name]
-                   @package = Package.get_by_project_and_name(params[:project_name], params[:package_name])
-                 elsif params[:project_name]
+    @watchable = if params[:project_name]
                    @project = Project.get_by_name(params[:project_name])
+                   params[:package_name] ? @package = Package.get_by_project_and_name(params[:project_name], params[:package_name]) : @project
```

Comment:
I think this method is a bit convoluted...
There's an `if ... elseif ... end` and a nested ternary, that is a hidden `if ... else`... There's too much going on here...
What about splitting setting `@package` and `@project`, then setting `@watchable`?.

Reply:
Do you mean like this?

```suggestion
                   @package = Package.get_by_project_and_name(params[:project_name], params[:package_name]) if params[:package_name]
                   @package || @project
```

Reply:
I changed the code to what we agreed offline, @danidoni. Please take a look.